frontend/lib/src/components/widgets/DataFrame/columns/cells/LinkCell.tsx

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from "react"
+
+import {
+  CustomCell,
+  CustomRenderer,
+  getMiddleCenterBias,
+  measureTextCached,
+  blend,
+  GridCellKind,
+} from "@glideapps/glide-data-grid"
+import UriOverlayEditor from "../../UriOverlayEditor"
+
+const UNDERLINE_OFFSET = 5
+
+export interface LinkCellProps {
+  readonly kind: "link-cell"
+  readonly link: {
+    readonly displayText: string
+    readonly href: string
+  }
+}
+
+export type LinkCell = CustomCell<LinkCellProps>
+
+function onClickSelect(
+  e: Parameters<NonNullable<CustomRenderer<LinkCell>["onSelect"]>>[0]
+):
+  | {
+      readonly displayText: string
+      readonly href: string
+    }
+  | undefined {
+  const canvas = document.createElement("canvas")
+  const ctx = canvas.getContext("2d", { alpha: false })
+  if (ctx === null) return
+
+  const { posX: hoverX, bounds: rect, cell, theme } = e
+  const font = `${theme.baseFontStyle} ${theme.fontFamily}`
+  ctx.font = font
+
+  const { link } = cell.data
+
+  const rectHoverX = rect.x + hoverX
+
+  const textWidth =
+    ctx.measureText(getLinkDisplayValue(link.href, link.displayText)).width +
+    theme.cellHorizontalPadding * 2
+
+  const isHovered = rectHoverX > rect.x && rectHoverX < rect.x + textWidth
+
+  if (isHovered) {
+    return link
+  }
+
+  return undefined
+}
+
+export function getLinkDisplayValue(
+  href: string,
+  displayText?: string
+): string {
+  if (!displayText) {
+    return href
+  }
+
+  try {
+    if (displayText.includes("(.*?)")) {
```

Comment:
I think this is probably not the best way to identify a valid regular expression. Surely there may be valid regex patterns to be used that would not have "(.*?)". What do you think @LukasMasuch ?

Reply:
Yeah. Should we just treat every non empty/null value as a potential regex? Then apply the "pattern" and if there is no match, display the display_text?

Reply:
This approach would mean we would allow any possible regex the user could come up with, which is good. But it would mean if they put in some invalid regex (or there is no match with one of the link cells), we will show the regex pattern as opposed to the url.

Reply:
resolved as we settled on using `(` and `)` to identify regex